text/3604-derive_c-enum_integer_conversions.md

```diff
@@ -0,0 +1,185 @@
+    //have to do bounds checking and change the function body to only match on fields with a value less than
+    //u{8,16,..}::MAX and more than u{8,16,...}::MIN
+    let value = value as usize;
+    match value { 
+        InputtedEnum::Field1 as usize => Ok(InputtedEnum::Field1),
+        InputtedEnum::FieldN as usize => Ok(InputtedEnum::FieldN),
+        none_of_those_fields => Err(()) //no fields have inputted value, conversion failed
+    }
+  }
+}
+```
+I\`m sure someone will be able to think of some cool optimizations for this, but this is plenty fast already
+
+## \#[derive(IntoInt)] implmentaion
+This one is really simple, its just using the `as` conversion inside the function, to convert the inputted enum to any number. But this allows it to be used inside generics where Into<{Integer}> is a trait bound, among others.
+```
+impl Into<{Integer}> for InputtedEnum {
+  fn into(self) -> {Integer} {
+    self as {Integer}
+  }
+}
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This feature is very insignificant, so thinking of any inherent, big drawbacks for it is a bit hard. It\`s main drawback is it isn\`t very important, might not be work the time of the rust team since C-style enums are not really used that much, perhaps due to their lack of documentation or that their usecases are pretty niche (making bitflags with unique names, etc.). I would be more than willing to write the derive macro and docs myself, since their isn\`t a lot of code to write. Maybe another drawback is that if not documented properly, developers might be confused why a TryFrom/Into implementation exists for their enum, since they never explicitly defined it (the derive macro is called TryFromInt, not TryFrom, etc..), which could cause a bit of confusion, but I think this can be avoided with good documentation
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+
+I don\`t think any other designs have been introduced/considered since again, C-style enums are very niche, but this problem is not very logically hard, unlike `Futures`, `const evaluation`, etc.. The impact of not having these helper macros is that it makes a niche part of the language harder to use than it needs to be, and forces the developer to write more boilerplate to get their code to work as intended. This functionality could and is done by a library like [strum](https://docs.rs/strum/latest/strum/), however, since this is a very simple and niche feature, I still think its in the scope of being added to the language, also I think since everyone who has used rust extensively knows of the uses and implications of TryFrom and Into traits, instead of some other 3rd-party library defined traits (e.g. FromRepr in the case of strum).
```

Comment:
What do you mean by "niche" here? Isn't being niche (i.e. very specialized), plus having a library alternative, exactly why this feature ***should not*** be standardized into the language?